index.js

```diff
@@ -113,7 +116,8 @@ module.exports = {
     }
 
     const app = this._findApp();
-    const preferNative = app._fetchBuildConfig.preferNative;
+    // If the polyfill is not included, we force the preferNative behavior
+    const preferNative = app._fetchBuildConfig.preferNative || this._checkSupports('fetch');
```

Comment:
I think we should respect `preferNative` over browser support and `alwaysIncludePolyfill`. The reason is people may be relying some xhr features and don't want native fetch even it exists.

Reply:
But if we do not include the polyfill, we need to set preferNative to true - otherwise, it will fail.
If we have two options: `preferNative` (default false) and `alwaysIncludePolyfill` (default true). 

Possible combinations:

`{ preferNative: true, alwaysIncludePolyfill: true }` --> polyfill is always included, but might not be used.

`{ alwaysIncludePolyfill: false }` --> In this case, the polyfill might be loaded or not loaded (it might either load fetch & abortcontroller polyfills, or only the abortcontroller polyfill, or none). If at least one polyfill is NOT loaded, then `preferNative` has to be `true`, otherwise the unpolyfilled parts will not work.

Reply:
I was saying the logic should be inverse: if the `preferNative` is set to `true`, then check second option `alwaysIncludePolyfill` and browser supports.

The behavior would be 
- `{ preferNative: true, alwaysIncludePolyfill: true }`  --> polyfills are always included, but might not be used
- `{ preferNative: true, alwaysIncludePolyfill: false }` --> conditionally include polyfills based on browser support
- `{ preferNative: false }` --> __skip__ checking `alwaysIncludePolyfill` and caniuse API request, always include polyfills

Reply:
Ah, got it.
I guess the question is, does it make sense for `alwaysIncludePolyfill` to default to true then, really? I mean, I guess it is more of an edge case to say: "I want to use the native polyfill when possible, but still always include the polyfill file"? Because now, to get this (IMHO most sensible) outcome, you need to set:

```
'ember-fetch': {
  preferNative: true,
  alwaysIncludePolyfill: false
}
```

Which feels a bit counter-intuitive - as these two options are pretty entangled and depend on each other in difficult to understand ways (e.g. setting `alwaysIncludePolyfill` does nothing if `preferNative=false`, ...). So I guess, wouldn't it be easier to understand to make `alwaysIncludePolyfill` default to false, so you can still, if needed for whatever edge-case reason, opt out of it, but the "happy path" is much simpler by just setting `preferNative: true`?

Reply:
Yea agree you're right about the ergonomics here. Previously I thought `caniuse-api` was making HTTP requests but after some digging I realized they are packing the dataset with the package. So the result of `preferNative: true` should be only the enhancement of reducing the redundant asset.
To reflect the opt-out path for edge-cases you mentioned, can you also update the README about the new API?